mhs/mhs/common/state/work_description.py

```diff
@@ -0,0 +1,122 @@
+        if persistence_store is None:
+            raise ValueError('Expected persistence store')
+        if not table_name:
+            raise ValueError('Expected non empty table name')
+
+        self.persistence_store = persistence_store
+        self.table_name = table_name
+
+        data = store_data[DATA]
+        self.message_key = store_data[DATA_KEY]
+        self.version = data[VERSION_KEY]
+        self.timestamp = data[TIMESTAMP]
+        self.status = data[STATUS]
+
+    async def publish(self):
+        logger.info('011', 'Attempting to publish work description {key}', {'key': self.message_key})
+        logger.info('012', 'Retrieving latest work description to check version')
+        latest_data = await self.persistence_store.get(self.table_name, self.message_key)
+        if latest_data is not None:
+            latest_version = latest_data[DATA][VERSION_KEY]
+            if latest_version > self.version:
+                logger.error('013', 'Failed to update message {key}, local version out of date',
+                             {'key': self.message_key})
+                raise OutOfDateVersionError(f'Failed to update message {self.message_key}: local version out of date ')
+
+        serialised = self._serialise_data()
+
+        old_data = await self.persistence_store.add(self.table_name, serialised)
+        logger.info('014', 'Successfully updated work description to state store for {key}', {'key': self.message_key})
+        return old_data
+
+    def _serialise_data(self):
+        data = {
+            DATA_KEY: self.message_key,
+            'DATA': {
+                TIMESTAMP: self.timestamp,
+                VERSION_KEY: self.version,
+                STATUS: self.status
+            }
+        }
+
+        return json.dumps(data)
+
+
+class WorkDescriptionFactory:
+
+    @staticmethod
+    async def get_work_description_from_store(persistence_store, table_name: str, key: str):
+        if persistence_store is None:
+            logger.error('001', 'Failed to get work description from store: persistence store is None')
+            raise ValueError('Expected non-null persistence store')
+        if key is None:
+            logger.error('002', 'Failed to get work description from store: key is None')
+            raise ValueError('Expected non-null key')
+
+        json_store_data = await persistence_store.get(table_name, key)
+        if json_store_data is None:
+            logger.error('003', 'Persistence store returned empty value for {key}', {'key': key})
+            raise EmptyWorkDescriptionError(f'Failed to find a value for key id {key}')
+
+        return WorkDescription(persistence_store, table_name, json_store_data)
+
+    @staticmethod
+    def create_new_work_description(persistence_store,
+                                    table_name: str,
+                                    key: str,
+                                    timestamp: str,
```

Comment:
I believe it's meant to be the last modified, yes it could be managed by this file but I figured I would allow complete control to the workflow manager as to the details of this state object. 

I can update it if you think it's preferential to have this updated automatically?

Reply:
I think it is better to have it updated automatically as we're going to have to update it somewhere, might as well be here.

Last modified or created? What about both? Probably a question for Dave.

Reply:
Do both